ARE/middleware/src/main/java/eu/asterics/mw/services/ResourceRegistry.java

```diff
@@ -638,20 +649,52 @@ public static String getResourceContentAsString(InputStream inputStream) throws
      * @throws FileNotFoundException 
      */
     public void storeResource(String data, String resourcePath, RES_TYPE resourceType) throws AREAsapiException, URISyntaxException, FileNotFoundException, IOException {
-        File filenameAsFile = ResourceRegistry.toFile(ResourceRegistry.getInstance().getResource(resourcePath, resourceType));
-        File parentDir = filenameAsFile.getParentFile();
+        storeResource(data, ResourceRegistry.getInstance().getResource(resourcePath, resourceType));
+    }
+    
+    /**
+     * Stores the given data at the given URI and encoded as {@link ResourceRegistry#ARE_FILE_ENCODING}. 
+     * @param data
+     * @param storeResourceURI
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    public static void storeResource(String data, URI storeResourceURI) throws URISyntaxException, IOException {
+        storeResource(data, ResourceRegistry.toFile(storeResourceURI));
+    }
+
+    /**
+     * Stores the given data at the given File object and encoded as {@link ResourceRegistry#ARE_FILE_ENCODING}. 
+     * @param data
+     * @param storeResourceFile
+     * @throws IOException
+     */
+    //Made this method private to not tempt developers to directly create File object paths and store to it.
+    private static void storeResource(String data, File storeResourceFile) throws IOException {
+        File parentDir = storeResourceFile.getParentFile();
         if (!parentDir.exists()) {
             if (!parentDir.mkdirs()) {
-                logger.severe("Could not create parent directories for data file: " + filenameAsFile);
-                throw new AREAsapiException("Could not create parent directories for data file.");
+                logger.severe("Could not create parent directories for data file: " + storeResourceFile);
+                throw new IOException("Could not create parent directories for data file.");
             }
         }
-        try (FileOutputStream outStream = new FileOutputStream(filenameAsFile);) {
-            IOUtils.write(data, outStream, ResourceRegistry.ARE_FILE_ENCODING);
-            outStream.flush();
-            outStream.close();
-        }
-    }    
+        storeResource(data, new FileOutputStream(storeResourceFile));
+    }
+    
+    /**
+     * Stores the given data to the given OutputStream and encoded as {@link ResourceRegistry#ARE_FILE_ENCODING}. 
```

Comment:
comment does not mention that output stream is closed.

Reply:
ok